### PR TITLE
dhparam generation

### PR DESCRIPTION
**Pull request by andytson-inviqa**
Apache/Nginx by default (through openssl defaults) use a weak 1024 bit dhparam generated at startup and only regenerated on reload or a new container.

Generating or loading an external dhparam file allows for this to be made a stronger 2048 bit.

Additionally scheduling regular daily dhparam regeneration ensures security.

This feature is off by default, and turned on using WEB_SSL_DHPARAM_ENABLE environment variable.

### CODE REVIEW

File: nginx/etc/confd/templates/supervisor/dhparam_regenerate.conf.tmpl [hunk 1]
```diff
@@ -0,0 +1,5 @@
+[program:dhparam-regenerate]
+command = supervisorctl --user supervisor --password supervisor start dhparam-regenerate
```
Comment (kierenevans) [inline nginx/etc/confd/templates/supervisor/dhparam_regenerate.conf.tmpl]: Will this be an infinite loop?

File: ubuntu/16.04/usr/local/share/bootstrap/common_functions.sh [hunk 2]
```diff
@@ -291,6 +291,13 @@ function test_remote_ports() {
   done
 }
 
+function supervisor_signal() {
+    local -r SIGNAL=$1
+    local -r PROGRAM_NAME=$2
+
+    supervisorctl --username supervisor --password supervisor signal "$SIGNAL" "$PROGRAM_NAME"
```
Comment (kierenevans) [inline ubuntu/16.04/usr/local/share/bootstrap/common_functions.sh]: Awesome!

File: php/shared/etc/confd/templates/supervisor/dhparam_regenerate.conf.tmpl [hunk 3]
```diff
@@ -0,0 +1,5 @@
+[program:dhparam-regenerate]
+command = supervisorctl --user supervisor --password supervisor start dhparam-regenerate
```
Comment (kierenevans) [inline php/shared/etc/confd/templates/supervisor/dhparam_regenerate.conf.tmpl]: Is this an infinite loop?

File: nginx/etc/confd/templates/supervisor/dhparam_regenerate.conf.tmpl (hunk 1 above)
Comment (andytson-inviqa) [reply nginx/etc/confd/templates/supervisor/dhparam_regenerate.conf.tmpl]: no autorestart and startsecs set to 0, so it wont restart under any condition.

Comment (andytson-inviqa) [reply nginx/etc/confd/templates/supervisor/dhparam_regenerate.conf.tmpl]: oh oops I see what you mean

Comment (andytson-inviqa) [reply nginx/etc/confd/templates/supervisor/dhparam_regenerate.conf.tmpl]: I tested it somehow, maybe before I "corrected" a non-issue in `container --silent dhparam_regenerate`

Comment (andytson-inviqa) [reply nginx/etc/confd/templates/supervisor/dhparam_regenerate.conf.tmpl]: fixed

File: php/shared/etc/confd/templates/supervisor/dhparam_regenerate.conf.tmpl (hunk 3 above)
Comment (andytson-inviqa) [reply php/shared/etc/confd/templates/supervisor/dhparam_regenerate.conf.tmpl]: fixed

**Comment by sroze** [comment]
Oups, conflicts :)

**Comment by andytson-inviqa** [comment]
@sroze fixed, just was one of those lines changed immediately after a changed line conflicts that could really auto-resolve but for some reason doesn't

**Pull request merged**